archive/schema.go

```diff
@@ -0,0 +1,140 @@
+package archive
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/brimsec/zq/pkg/fs"
+)
+
+const configName = "zar.json"
+
+var DefaultConfig Config = Config{
+	Version:          0,
+	LogSizeThreshold: 500 * 1024 * 1024,
+}
+
+type Config struct {
+	Version          int `json:"version"`
+	LogSizeThreshold int `json:"log_size_threshold"`
+}
+
+func (c *Config) Write(path string) error {
+	b, err := json.Marshal(c)
+	if err != nil {
+		return err
+	}
+	f, err := ioutil.TempFile(filepath.Dir(path), "."+configName+".*")
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(b)
+	if err != nil {
+		f.Close()
+		return err
+	}
+	err = f.Close()
+	if err != nil {
+		return err
+	}
+	return os.Rename(f.Name(), path)
+}
+
+func ConfigRead(path string) (*Config, error) {
+	f, err := fs.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+	var c Config
+	err = json.Unmarshal(b, &c)
+	if err != nil {
+		return nil, err
+	}
+	return &c, nil
+}
+
+type CreateFlags struct {
+	megaThresh int
+	byteThresh int
+}
+
+func (f *CreateFlags) SetFlags(fs *flag.FlagSet) {
```

Comment:
I am skeptical of this pattern of specifying flags in library packages. It is nice to look and the command definition and see all flags specified whereas this pattern obscures them. Additionally it seems like there is a high likelihood of flag name collisions.

Reply:
I'm not a fan either.

Reply:
Ok - I started with this by having a 'CreateOptions' struct:
```go
type CreateOptions struct {
   LogSizeThreshold *int
}
```
Where individual values, if set, specify overrides to the default. Do yall find that more palatable?